pkg/v1/cli/command/core/context.go

```diff
@@ -0,0 +1,643 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package core
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/aunum/log"
+	"github.com/spf13/cobra"
+	"golang.org/x/oauth2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	cliv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/cli/v1alpha1"
+	configv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/config/v1alpha1"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/auth/csp"
+	tkgauth "github.com/vmware-tanzu/tanzu-framework/pkg/v1/auth/tkg"
+	wcpauth "github.com/vmware-tanzu/tanzu-framework/pkg/v1/auth/wcp"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/component"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/pluginmanager"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/config"
+)
+
+var (
+	stderrOnly, forceCSP, staging, onlyCurrent                                  bool
+	ctxName, ctxType, endpoint, apiToken, kubeConfig, kubeContext, getOutputFmt string
+)
+
+const (
+	knownGlobalHost = "cloud.vmware.com"
+)
+
+var contextCmd = &cobra.Command{
+	Use:     "context",
+	Short:   "Configure and manage contexts for the Tanzu CLI",
+	Aliases: []string{"ctx", "contexts"},
+	Annotations: map[string]string{
+		"group": string(cliv1alpha1.SystemCmdGroup),
+	},
+}
+
+func init() {
+	contextCmd.SetUsageFunc(cli.SubCmdUsageFunc)
+	contextCmd.AddCommand(
+		createCtxCmd,
+		listCtxCmd,
+		getCtxCmd,
+		deleteCtxCmd,
+		useCtxCmd,
+	)
+
+	initCreateCtxCmd()
+
+	listCtxCmd.Flags().StringVarP(&ctxType, "type", "t", "", "context type (k8s|tmc)")
+	listCtxCmd.Flags().BoolVar(&onlyCurrent, "current", false, "list only current active contexts")
+	listCtxCmd.Flags().StringVarP(&outputFormat, "output", "o", "table", "output format: table|yaml|json")
+
+	getCtxCmd.Flags().StringVarP(&getOutputFmt, "output", "o", "yaml", "output format: yaml|json")
+
+	deleteCtxCmd.Flags().BoolVarP(&unattended, "yes", "y", false, "delete the context entry without confirmation")
+}
+
+var createCtxCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create a Tanzu CLI context",
+	RunE:  createCtx,
+	Example: `
+	# Create a TKG management cluster context using endpoint
+	tanzu context create --endpoint "https://k8s.example.com" --name mgmt-cluster
+
+	# Create a TKG management cluster context using kubeconfig path and context
+	tanzu context create --kubeconfig path/to/kubeconfig --kubecontext kubecontext --name workload-cluster
```

Comment:
nit: the name `workload-cluster` is going to be confusing, maybe just mgmt-cluster?

Reply:
Fixed